Realm/RLMSyncConfiguration.h

```diff
@@ -41,16 +43,76 @@ typedef NS_ENUM(NSUInteger, RLMClientResetMode) {
     /// re-downloaded Realm will initially contain only the data present at the time the Realm
     /// was backed up on the server.
     ///
-    /// - see: `rlmSync_clientResetBackedUpRealmPath` or `SyncError.clientResetInfo()` for more information on accessing the recovery directory.
-    RLMClientResetModeManual,
+    /// @see: `rlmSync_clientResetBackedUpRealmPath` and `RLMSyncErrorActionToken` for more information on accessing the recovery directory and error information.
+    ///
+    /// The manual client reset mode handler can be set in two places:
+    ///  1. As an ErrorReportingBlock argument at `RLMSyncConfiguration.manualClientResetHandler`.
+    ///  2. As an ErrorReportingBlock in the `RLMSyncManager.errorHandler` property.
+    ///  @see: `RLMSyncManager.errorHandler`
+    ///
+    ///  When an `RLMSyncErrorClientResetError` is thrown, the following rules determine which block is executed:
+    ///  - If an error reporting block is set in `.manualClientResetHandler` and the `RLMSyncManager.errorHandler`, the `.manualClientResetHandler` block will be executed.
+    ///  - If an error reporting block is set in either the `.manualClientResetHandler` or the `RLMSyncManager`, but not both, the single block will execute.
+    ///  - If no block is set in either location, the client reset will not be handled. The application will likely need to be restarted and unsynced local changes may be lost.
+    /// @note: The `RLMSyncManager.errorHandler` is still invoked under all `RLMSyncError`s *other than* `RLMSyncErrorClientResetError`. @see `RLMSyncError` for an exhaustive list.
+    RLMClientResetModeManual = 0,
     /// All unsynchronized local changes are automatically discarded and the local state is
     /// automatically reverted to the most recent state from the server. Unsynchronized changes
     /// can then be recovered in the post-client-reset callback block.
     ///
-    /// If RLMClientResetModeDiscardLocal is enabled but the client reset operation is unable to complete
+    /// If `RLMClientResetModeDiscardLocal` is enabled but the client reset operation is unable to complete
     /// then the client reset process reverts to manual mode. Example) During a destructive schema change this
     /// mode will fail and invoke the manual client reset handler.
-    RLMClientResetModeDiscardLocal
+    ///
+    /// The RLMClientResetModeDiscardLocal mode supports two client reset callbacks -- `RLMClientResetBeforeBlock`, `RLMClientResetAfterBlock` -- which can be passed as arguments when creating the `RLMSyncConfiguration`.
+    ///- see: `RLMClientResetAfterBlock` and `RLMClientResetBeforeBlock`
```

Comment:
Some `- see`s which should be `@see`, and Jazzy needs a blank line before it to render well.

Linking to other documented symbols in DocC requires surrounding the name in double backticks. We haven't gone through and updated all the existing docs for this, but anything new should do it.

Reply:
Do these go for swift files also?

Reply:
The double back tick, that is.

Reply:
yes